api/tests/test_canteens.py

```diff
@@ -663,23 +697,119 @@ def test_get_canteen_actions(self):
             self.assertEqual(returned_canteens[index]["id"], canteen.id)
             self.assertEqual(returned_canteens[index]["action"], action)
 
+    @override_settings(ENABLE_TELEDECLARATION=True)
+    @authenticate
+    def test_get_actions_missing_data(self):
+        """
+        Even if the diagnostic is complete, the mandatory information on the canteen level should
+        return a "complete_diagnostic"
+        """
+        # First a case in which the canteen is complete
+        canteen = CanteenFactory.create(
+            production_type=Canteen.ProductionType.ON_SITE,
+            publication_status=Canteen.PublicationStatus.PUBLISHED,
+            management_type=Canteen.ManagementType.DIRECT,
+            yearly_meal_count=1000,
+            daily_meal_count=12,
+            siret="96766910375238",
+            city_insee_code="69123",
+        )
+        DiagnosticFactory.create(year=2021, canteen=canteen, value_total_ht=1000)
+        canteen.managers.add(authenticate.user)
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "40_teledeclare")
+
+        # If mandatory data on the canteen is missing, we need a 35_fill_canteen_data
+        canteen.yearly_meal_count = None
+        canteen.save()
+
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "35_fill_canteen_data")
+
+        # Central cuisines should have the number of satellites filled in
+        canteen.yearly_meal_count = 1000
+        canteen.production_type = Canteen.ProductionType.CENTRAL
+        canteen.satellite_canteens_count = None
+        canteen.save()
+
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "35_fill_canteen_data")
+
+        canteen.satellite_canteens_count = 123
+        canteen.save()
+
+        response = self.client.get(reverse("list_actionable_canteens", kwargs={"year": 2021}))
+        returned_canteens = response.json()["results"]
+        self.assertEqual(returned_canteens[0]["action"], "40_teledeclare")
```

Comment:
Future PR: should satellite count === count of satellites in db to be able to TD ?